java-spiffe-core/src/main/java/io/spiffe/workloadapi/StreamObservers.java

```diff
@@ -45,7 +46,9 @@ public void onNext(final Workload.X509SVIDResponse value) {
 
             @Override
             public void onError(final Throwable t) {
-                log.log(Level.SEVERE, "X.509 context observer error", t);
+                if (!t.getMessage().contains(CANCELLED)) {
```

Comment:
For all these cases where we try to detect gRPC `CANCELLED` status codes, can we use the Status API from grpc-java to parse the exception for us and make the decision based on the Status enum value? https://github.com/grpc/grpc-java/blob/7aa5598dc04ef72438f62071ecaa84d781c86b6a/api/src/main/java/io/grpc/Status.java#L394-L401

Reply:
Nice! Thanks